# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1751498765 -7200
# Parent  8c92df57a192c5a2a163667ea2dcce3c94026fd7
NNN26 - Add the following to CSP of compose window, see bug 1964799:
  script-src moz-extension:
  img-src data:
  worker-src blob:

This is necessary for WebExtension *Experiments* where
"content_security_policy": "default-src 'self'; script-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self' data:; worker-src 'self' blob:"
in the manifest doesn't work. The console message clearly shows that CSP is taken from the compose window itself, and that can't be modified
via the extension manifest or dynamically from within the extension.

diff --git a/mail/components/compose/content/messengercompose.xhtml b/mail/components/compose/content/messengercompose.xhtml
--- a/mail/components/compose/content/messengercompose.xhtml
+++ b/mail/components/compose/content/messengercompose.xhtml
@@ -42,9 +42,9 @@
 <head>
   <title>&msgComposeWindow.title;</title>
   <meta
     http-equiv="Content-Security-Policy"
-    content="default-src chrome:; script-src chrome: resource: 'unsafe-inline'; style-src chrome: 'unsafe-inline'; img-src chrome: moz-icon:; worker-src resource:; connect-src http: https:"
+    content="default-src chrome:; script-src chrome: resource: moz-extension: 'unsafe-inline'; style-src chrome: 'unsafe-inline'; img-src chrome: moz-icon: data:; worker-src resource: blob:; connect-src http: https:"
   />
   <link
     rel="stylesheet"
     href="chrome://messenger/skin/messengercompose/messengercompose.css"
